File: striver/S3L1/union.py
# def merge(larr,rarr):
#      result = [0]*(len(larr)+len(rarr))
#      k = int(0)
#      l = int(0)
#      r = int(0)
#      while l<len(larr) and r < len(rarr):
#          if larr[l]>=rarr[r]:
#              result[k]=rarr[r]
#              k+=1
#              r+=1
#          else:
#              result[k] = larr[l]
#              k+=1
#              l+=1  
#      while l<len(larr):
#          result[k] = larr[l]
#          l+=1
#          k+=1
#      while r<len(rarr):
#          result[k] = rarr[r]
#          r+=1
#          k+=1
#      return result
# def removeDuplicates(nums):
#     fr = 0
#     sr = 0
#     while fr<len(nums):
#         if nums[sr] != nums[fr]:
#             sr+=1
#             nums[sr] = nums[fr]
#         fr+=1
#     return sr+1
# size1 = int(input())
# larr = [0]*size1
# for i in range(0,size1):
#     larr[i]=int(input())
# size2 = int(input())
# rarr = [0]*size2
# for i in range(0,size2):
#     rarr[i]=int(input())
# result = merge(larr,rarr)
# size = removeDuplicates(result)
# for i in range(0,size):
#     print(result[i],end=" ")


size1 = int(input())
larr = [0]*size1
for i in range(0,size1):
    larr[i]=int(input())
size2 = int(input())
rarr = [0]*size2
for i in range(0,size2):
    rarr[i]=int(input())
hs = set()
for i in range(0,size1):
    hs.add(larr[i])
for i in range(0,size2):
    hs.add(rarr[i])
print(hs)

File: striver/S3L1/union.java
package S3L1;

import java.util.HashSet;
import java.util.Scanner;

public class union {
    // public static void main(String[] args) {
    //     Scanner sc = new Scanner(System.in);
    //     int size1 = sc.nextInt();
    //     int[] larr = new int[size1];
    //     for(int i=0;i<size1;i++){
    //         larr[i]=sc.nextInt();
    //     }
    //     int size2 = sc.nextInt();
    //     int[] rarr = new int[size2];
    //     for(int i=0;i<size2;i++){
    //         rarr[i]=sc.nextInt();
    //     }
    //     int[] result = merge(larr,rarr);
    //     int size = removeDuplicates(result);
    //     for(int i=0;i<size;i++){
    //         System.out.print(result[i]+" ");
    //     }
    // }
    // public static int[] merge(int[] larr,int[] rarr){
    //     int[] result = new int[larr.length+rarr.length];
    //     int k =0;
    //     int l =0;
    //     int r = 0;
    //     while(l<larr.length && r<rarr.length){
    //         if(larr[l]>=rarr[r]){
    //             result[k]=rarr[r];
    //             k++;
    //             r++;
    //         }else {
    //             result[k]=larr[l];
    //             k++;
    //             l++;
    //         }
    //     }
    //     while(l<larr.length){
    //         result[k]=larr[l];
    //         k++;
    //         l++;
    //     }
    //     while(r<rarr.length){
    //         result[k]=rarr[r];
    //         k++;
    //         r++;
    //     }
    //     return result;
    // }
    // public static int removeDuplicates(int[] nums) {
    //     int fr = 0;
    //     int sr = 0;
    //     while(fr<nums.length){
    //         if(nums[sr] != nums[fr]) {
    //             nums[++sr] = nums[fr];
    //         }
    //         fr++;
    //     }
    //     return ++sr;
    // }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int size1 = sc.nextInt();
        int[] larr = new int[size1];
        for(int i=0;i<size1;i++){
            larr[i]=sc.nextInt();
        }
        int size2 = sc.nextInt();
        int[] rarr = new int[size2];
        for(int i=0;i<size2;i++){
            rarr[i]=sc.nextInt();
        }
        HashSet<Integer> hs = new HashSet<>();
        for(int i=0;i<size1;i++){
            hs.add(larr[i]);
        }
        for(int i=0;i<size2;i++){
            hs.add(rarr[i]);
        }
        System.out.println(hs);
    }
}
